### PR TITLE
[wasm] Generate jiterpreter feature detect modules dynamically

### LINKED ISSUE #3442: Scan for binaries is failing from wasm file in runtime repo
The CI build shows a warning in the `Scan for binaries` task:

```
./.dotnet/dotnet darc vmr scan-binary-files --vmr "/mnt/vss/_work/1/s" --tmp "/mnt/vss/_work/_temp" --baseline-file "src/VirtualMonoRepo/allowed-binaries.txt" || (echo '##[error]Found binaries in the VMR' && exit 1)
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /mnt/vss/_work/_temp/e45a5dcc-b89a-4f39-8e02-67abfa651e62.sh
src/runtime/src/mono/wasm/runtime/wasm-simd-feature-detect.wasm
##[error]Found binaries in the VMR
##[error]Bash exited with code '1'.
Finishing: Scan for binaries
```

### PR DESCRIPTION
**Pull request by kg**
Fixes https://github.com/dotnet/source-build/issues/3442

### CODE REVIEW

File: src/mono/wasm/runtime/jiterpreter-feature-detect.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        export: true,
+        locals: {}
+    }, () => {
+        builder.i32_const(0);
+        builder.appendSimd(WasmSimdOpcode.i32x4_splat);
+        builder.appendU8(WasmOpcode.drop);
+        builder.appendU8(WasmOpcode.end);
+    });
+    // Magic number and version
+    builder.appendU32(0x6d736100);
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/jiterpreter-feature-detect.ts:64]: do method names like `appendU32` minify well with rollup ?

Comment (kg) [reply src/mono/wasm/runtime/jiterpreter-feature-detect.ts:64]: I'm not sure what you mean

Comment (pavelsavara) [reply src/mono/wasm/runtime/jiterpreter-feature-detect.ts:64]: how many bytes we are adding to size of JS with this PR ?

File: src/mono/wasm/runtime/jiterpreter-support.ts [hunk 2]
```diff
@@ -482,13 +482,16 @@ export class WasmBuilder {
         return result;
     }
 
-    _generateImportSection () {
+    _generateImportSection (includeFunctionTable?: boolean) {
```
Comment (kotlarmilos) [inline src/mono/wasm/runtime/jiterpreter-support.ts:485]: I recommend standardizing the annotation of optional parameters -- either implicitly using the `?` symbol or explicitly using `| undefined`.

**Review by kotlarmilos — COMMENTED** [review]
LGTM!

File: src/mono/wasm/runtime/jiterpreter-feature-detect.ts (hunk 1 above)
Comment (kg) [reply src/mono/wasm/runtime/jiterpreter-feature-detect.ts:60]: each feature detect generator is around 400 bytes of uncompressed js after minification, and i'm pretty sure brotli will crunch it up just fine. we had a decent amount of code before this that it's replacing, so i think it's ok

Comment (pavelsavara) [reply src/mono/wasm/runtime/jiterpreter-feature-detect.ts:60]: https://github.com/dotnet/runtime/issues/86447

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**